### ISSUE TITLE
Cable trace of multiple pairs of MPO CABLE

### ISSUE THREAD

**Issue by surlyjake** [netbox-community/netbox#3772]
### Environment
Netbox Version 2.6.7

 
```

  Device A                             Patch Panel A                              Patch Panel B          

+-----------+                +-------------+                                              +------------+

| Interface | --- Cable1 --- | Front Port1 | --+                                       +--| Front Port1|  ----+

+-----------+                +-------------+   |                                       |  +------------+      |

                                               |                                       |                      |

                                               |                                       |                      |

                                               +-----------+               +-----------+

                                               | Rear Port1| --- Cable3 ---| Rear Port1|                   Cable4

                                               +-----------+     (MPO)     +-----------+

                                               |                                       |                      |

  Device B                                     |                                       |                      |

+-----------+                +-------------+   |                                       |  +-------------+     |

| Interface | --- Cable2 --- | Front Port2 | --+                                       +--| Front Port 2|  ---+

+-----------+                +-------------+                                              +-------------+

```
 
Setup the environment for the topology:

1. Create a device type for the endpoint device "example box"
2. While viewing the "example box" device type, select "add components", then select "interfaces".
3. Add a SFP interface ("test int") to the "example box" device type.
4. Create a device type for a patch panel. ("example patch")
5. While viewing the "example patch" device type, select "add components", then "rear port". Name: "rear", type: "MPO", positions: "12"
6. While viewing the "example patch" device type, select "add components", then "front port". Name: "front[1-12]", type: "LC", rear ports: "back:1" hold down shift, then select "back:12"
7. Click "devices", then the "add" icon next to "devices". Name: "example box 1". Device role "server". Device type: "example box".
8. Click "devices", then the "add" icon next to "devices". Name: "example box 2". Device role "server". Device type: "example box".
9. Click "devices", then the "add" icon next to "devices". Name: "example patch 1". Device role "Patch Panel. Device type: "example patch".
10. Click "devices", then the "add" icon next to "devices". Name: "example patch 2". Device role "Patch Panel. Device type: "example patch".
11. Click "devices" then "devices".
12. Select "example box 1". In the interface list, for interface "test int", select the green connection icon (second from the left). Select "front port".
13. On the "B side", select device "example patch 1" and port name "front1"
14. Click "devices" then "devices".
15. Select "example box 2". In the interface list, for interface "test int", select the green connection icon (second from the left). Select "front port".
16. On the "B side", select device "example patch 1" and port name "front2"
17. Click "devices" then "devices".
18. Select "example patch 1". In the rear ports list, for port "back", select the green connection icon. Select "rear port"
19. On the "B side", select "example patch 2" and port name "rear".
20. Click "devices" then "devices".
21. Select "example patch 2". In the front ports list, for port "front1". Select the green connection icon. Select "front port"
22. On the "B side", select "example patch 2" and port name "front2"

    

### Expected Behavior

Expected to see the observed cables AND the following:
  d. Cable from "example patch 2" rear port "rear" => "example patch 1" rear port "rear"
  e. Cable from "example patch 1" front port "front2" => "example box 2" interface "test int"
 

### Observed Behavior

Now to check the topology:
1. Select "devices", then "devices".
2. Select "example box 1".
  a. In the interfaces list, it correctly shows that "example box 1" interface "test int" is connected to "example box 2" interface "test int".
3. Select the blue icon next to the cable used in "test int" interface.
4. Cable trace shows:
  a. Cable from "example box 1" interface "test int" => "example patch 1" front port "front1"
  b. Cable from "example patch 1" rear port "back" => "example patch 2" rear port "back"
  c. Cable from "example patch 2" front port "front1" => "example patch 2" front port "front2".
5. From the cable trace, the last two cables of the full path are missing.

**Comment by jeremystretch** [comment]
Please expand your bug report to include the specific steps you are taking to reproduce the behavior.

**Comment by surlyjake** [comment]
Setup the environment for the topology:

1.       Create a device type for the endpoint device “example box”

2.       While viewing the “example box” device type, select “add components”, then select “interfaces”.

3.       Add a SFP interface (“test int”) to the “example box” device type.

4.       Create a device type for a patch panel. (“example patch”)

5.       While viewing the “example patch” device type, select “add components”, then “rear port”. Name: “rear”, type: “MPO”, positions: “12”

6.       While viewing the “example patch” device type, select “add components”, then “front port”. Name: “front[1-12]”, type: “LC”, rear ports: “back:1” hold down shift, then select “back:12”

7.       Click “devices”, then the “add” icon next to “devices”. Name: “example box 1”. Device role “server”. Device type: “example box”.

8.       Click “devices”, then the “add” icon next to “devices”. Name: “example box 2”. Device role “server”. Device type: “example box”.

9.       Click “devices”, then the “add” icon next to “devices”. Name: “example patch 1”. Device role “Patch Panel. Device type: ”example patch”.

10.   Click “devices”, then the “add” icon next to “devices”. Name: “example patch 2”. Device role “Patch Panel. Device type: ”example patch”.

11.   Click “devices” then “devices”.

12.   Select “example box 1”. In the interface list, for interface “test int”, select the green connection icon (second from the left). Select “front port”.

13.   On the “B side”, select device “example patch 1” and port name “front1”

14.   Click “devices” then “devices”.

15.   Select “example box 2”. In the interface list, for interface “test int”, select the green connection icon (second from the left). Select “front port”.

16.   On the “B side”, select device “example patch 1” and port name “front2”

17.   Click “devices” then “devices”.

18.   Select “example patch 1”. In the rear ports list, for port “back”, select the green connection icon. Select “rear port”

19.   On the “B side”, select “example patch 2” and port name “rear”.

20.   Click “devices” then “devices”.

21.   Select “example patch 2”. In the front ports list, for port “front1”. Select the green connection icon. Select “front port”

22.   On the “B side”, select “example patch 2” and port name “front2”

 

Now to check the topology:

1.       Select “devices”, then “devices”.

2.       Select “example box 1”.

a.       In the interfaces list, it correctly shows that “example box 1” interface “test int” is connected to “example box 2” interface “test int”.

3.       Select the blue icon next to the cable used in “test int” interface.

4.       Cable trace shows:

a.       Cable from “example box 1” interface “test int” => “example patch 1” front port “front1”

b.       Cable from “example patch 1” rear port “back” => “example patch 2” rear port “back”

c.       Cable from “example patch 2” front port “front1” => “example patch 2” front port “front2”.

5.       From the cable trace, the last two cables of the full path are missing. Expected to see the above cables AND the following:

a.       Cable from “example patch 2” rear port “rear” => “example patch 1” rear port “rear”

b.       Cable from “example patch 1” front port “front2” => “example box 2” interface “test int”

**Comment by jeremystretch** [comment]
@surlyjake Please update the body of the issue with the steps to reproduce and fix the formatting so that they are legible.

**Comment by surlyjake** [comment]
Sorry for formatting. Updated the body. Should be legible now.

**Comment by fablabo** [comment]
This is most likely to be resolved by implementing the suggested changes from #3633

**Comment by jeremystretch** [comment]
Closing this as it will be addressed by #4388.
